hutch_python/calc_defaults.py

```diff
@@ -31,9 +31,23 @@ def collect_functions(modules):
                 pass
     return HelpfulNamespace(**functions)
 
+# import specific function to have at the top level of the namespace
+try:
+    from pcdscalc.diffraction import (bragg_angle, darwin_width)
+except ImportError:
+    print("Failed to import functions from pcdscalc.diffraction")
+
+try:
+    from pcdscalc.xray import transmission
+except ImportError:
+    print("Failed to import functions from pcdscalc.xray")
```

Comment:
This won't really work, because `transmission` will be undefined below

Plus, you wouldn't really want to see `transmission` as `None` or some stub function when you do `calc.transmission`, would you?

Maybe something like:
```python
try:
    from pcdscalc.xray import transmission
    optional_functions = {"transmission": transmission}
except ImportError:
    ...

calc_namespace = HelpfulNamespace(
    ...,
    **optional_functions
)
```

Reply:
I think this is definitely the cleanest way to handle it (make an optional_functions dict, import and insert into it as possible, put whatever you get into the final namespace)

Reply:
Yeah that is much better indeed. I realized afterwards that my try / except as coded now would not do anything here. Good suggestion on how to fix that!